### PR TITLE
Add Equivalent Java Examples (#19)

### PR DESCRIPTION
**Pull request by RCHowell**
https://github.com/partiql/partiql-lang-kotlin/issues/19

*Description of changes:*
Adds an equivalent java example for ParserExample and EvaluationWithBindings

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/src/java/org/partiql/examples/EvaluationJavaExample.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import org.partiql.lang.eval.ExprValue;
+import org.partiql.lang.eval.Expression;
+import org.partiql.lang.syntax.SqlParser;
+
+import java.io.PrintStream;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonMap;
+
+public class EvaluationJavaExample extends Example {
```
Comment (am357) [inline examples/src/java/org/partiql/examples/EvaluationJavaExample.java:34]: For this and others that apply: would be great to have a JavaDoc explaining the example.

File: examples/src/java/org/partiql/examples/EvaluationJavaExample.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        final Expression e = pipeline.compile(query);
+
+        // This is the value of the global variable.
+        final String userName = "Homer Simpson";
+        final ExprValue usernameValue = pipeline.getValueFactory().newString(userName);
+
+        // [Bindings.ofMap] can be used to construct a [Bindings] instance of
+        // bindings with previously materialized values.
+        final Bindings<ExprValue> globalVariables = Bindings.ofMap(singletonMap("user_name", usernameValue));
+        print("global variables:", "user_name => userName");
```
Comment (am357) [inline examples/src/java/org/partiql/examples/EvaluationJavaExample.java:61]: Should we print the `globalVariables`?

**Review by am357 — COMMENTED** [review]
LGTM, just two minor comments.

File: examples/src/java/org/partiql/examples/EvaluationJavaExample.java (hunk 2 above)
Comment (RCHowell) [reply examples/src/java/org/partiql/examples/EvaluationJavaExample.java:61]: Yes. This is what was in the Kotlin example. I will fix both.

File: examples/src/java/org/partiql/examples/EvaluationJavaExample.java (hunk 1 above)
Comment (RCHowell) [reply examples/src/java/org/partiql/examples/EvaluationJavaExample.java:34]: Ack

**Comment by RCHowell** [comment]
```
> $ ./gradlew :examples:run --args="EvaluationJavaExample"                                                                                                                             

> Task :examples:run
Running example: EvaluationJavaExample
PartiQL query:
    'Hello, ' || user_name
global variables:
    user_name => 'Homer Simpson'
result
    'Hello, Homer Simpson'
End of example: EvaluationJavaExample

> $ ./gradlew :examples:run --args="ParserJavaExample"                                                                                                                                  

> Task :examples:run
Running example: ParserJavaExample
PartiQL query
    SELECT exampleField FROM exampleTable WHERE anotherField > 10
Serialized AST

    (
      query
      (
        select
        (
          project
          (
# .......
End of example: ParserJavaExample
```

**Pull request merged**